packages/ui-editable/src/Editable/props.ts

```diff
@@ -21,71 +21,111 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-
+import React from 'react'
 import PropTypes from 'prop-types'
 
 import type { PropValidators } from '@instructure/shared-types'
 
-type EditableOwnProps = {
+type GetContainerProps = (props?: Record<string, any>) => {
+  ref: React.RefCallback<any>
+  onMouseOver: (event: React.MouseEvent) => void
+  onMouseOut: (event: React.MouseEvent) => void
+  onMouseDown?: (event: React.MouseEvent) => void
+  onKeyUp?: (event: React.KeyboardEvent) => void
+  readOnly?: boolean
+} & Record<string, any>
+
+type GetViewerProps = (props?: Record<string, any>) => {
   mode: 'view' | 'edit'
-  onChangeMode: (...args: any[]) => any
-  children?: (...args: any[]) => any
-  render?: (...args: any[]) => any
-  value?: any
-  onChange?: (...args: any[]) => any
   readOnly?: boolean
-  elementRef?: (element: Element | null) => void
-}
+} & Record<string, any>
 
-type PropKeys = keyof EditableOwnProps
+type GetEditorProps = (props?: Record<string, any>) => {
+  mode: 'view' | 'edit'
+  onBlur: () => void
+  editorRef: (el: HTMLElement | null) => void
+  readOnly?: boolean
+} & Record<string, any>
 
-type AllowedPropKeys = Readonly<Array<PropKeys>>
+type GetEditButtonProps = (props?: Record<string, any>) => {
+  onClick: () => void
+  onFocus: () => void
+  onBlur: () => void
+  isVisible: boolean
+  buttonRef: (el: Element) => void
+  readOnly?: boolean
+} & Record<string, any>
 
-type EditableProps = EditableOwnProps
+type EditableRenderProps = {
+  mode: string
+  getContainerProps: GetContainerProps
+  getViewerProps: GetViewerProps
+  getEditorProps: GetEditorProps
+  getEditButtonProps: GetEditButtonProps
+}
 
-const propTypes: PropValidators<PropKeys> = {
+type EditableOwnProps = {
   /**
    * If `'view'`: the view component is rendered,
    * if `'edit'`: the edit component is rendered
    */
-  mode: PropTypes.oneOf(['view', 'edit']).isRequired,
+  mode: 'view' | 'edit'
   /**
    * Called when the component's mode changes
    * @param {string} new_mode
    */
-  onChangeMode: PropTypes.func.isRequired,
-
+  onChangeMode: (newMode: string) => void
   /**
    * @param {Object} renderProps
+   *
    * @param {Boolean} mode
+   *
    * @param {Function} renderProps.getContainerProps - Props to be spread onto the container element
+   *
    * @param {Function} renderProps.getEditorProps - Props to be spread onto the editor element
+   *
    * @param {Function} renderProps.getEditButtonProps - Props to be spread onto the edit button element
    */
```

Comment:
this looks quite ugly in the docs with the new TS props, I'd refactor it to:
```
  /**
   * Function that you can supply that will return the children of this component.
   * It has one parameter has the following fields:
   * - mode: `view` or `edit`, depending on whether the view or the editor should be rendered.
   * - getContainerProps(props) - Props to be spread onto the container element
   * - getEditorProps(props) - Props to be spread onto the editor element
   * - getEditButtonProps(props) - Props to be spread onto the edit button element
   */
```

Reply:
(same for the other fields, please remove any JSDoc annotations (like `@param`)and use Markdown if needed, since they look ugly